### ISSUE TITLE
VariantManagament: Unable to switch to default view after creating new one

### ISSUE THREAD

**Issue by superdyzio** [SAP/ui5-webcomponents-react#6616]
### Describe the bug

On some of our views we have analytical tables with VariantManagament. Whenever user enters such view for the first time he/she gets the default, hardcoded variant called `Standard`. When user introduces some change in the column config then `Save as` button becomes active, user saves the view as `Standard2` (name doesn't matter here) with `Set as default` checkbox checked. After this action user gets to the `Standard2` variant and it's not possible to change the variant back to `Standard`. Also, it looks like `Standard` is still selected.

In the scenario when user have more variants, only the previously selected one is blocked - other variants can be accessed and when user switches to one of them then `Standard` would get unblocked.

### Reproduction steps

Prerequisite: only default variant present in variant manager

1. Introduce a change
2. `Save as` new variant
3. Try to navigate back to previous variant (doesn't work)
4. Introduce next change
5. Create third variant
6. Try to navigate to second variant (doesn't work)
7. Try to navigate to first variant (works)
8. Try to navigate to second variant (now works)

### Expected Behaviour

All of the variants are selectable and working no matter if user just created a new variant.

### Screenshots or Videos

![image](https://github.com/user-attachments/assets/470169a4-1a2e-4a19-80ed-7cdb5cb37084)

### UI5 Web Components for React Version

2.1.1

### UI5 Web Components Version

2.2.0

### Browser

Chrome

### Operating System

MacOS Sequoia 15.1

### Additional Context

Analytical tables used with the VariantManagament are very big, up to almost 300 columns and god-knows-how-many rows.

### Organization

SAP Fioneer

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by superdyzio** [comment]
@Lukas742 I used your [example](https://stackblitz.com/edit/ui5wcr-vm-saveas-9wgvec?file=src%2FApp.tsx) and was able to reproduce it and I'm a bit confused - I didn't change a thing in the stackblitz example code

https://github.com/user-attachments/assets/f546c11a-7ce7-4756-9954-235a9386e956

**Comment by Lukas742** [comment]
@superdyzio there was a mistake in my example. I didn't update the `isDefault` property of the first `VariantItem`, which causes multiple default variants after saving ([not supported](https://experience.sap.com/fiori-design-web/variant-management/#default-view)). Here's the updated example:

https://stackblitz.com/edit/ui5wcr-vm-saveas-lnktd4?file=src%2FApp.tsx

Could you check again if this works for you now?

[This PR](https://github.com/SAP/ui5-webcomponents-react/pull/6641) will add a note explaining which props need to be unique for each `VariantManagement` and will also fix some more issues with internal `boolean` value assignments.

**Comment by superdyzio** [comment]
@Lukas742 I see, we'll make sure that we always have only one default variant, thanks for the investigation! we're waiting for your remaining fixes and looking forward to the release including our recent bugfixes 🎉

**Comment by Lukas742** [comment]
The issue I found previously will also be resolved by the linked PR. In some cases the `isDefault` value was not reflected correctly in the manage or save view dialog. 

Apart from that I've edited my example from above to include all necessities:
- The `isDefault` prop is now updated accordingly.
- Selection change is now handled by the React state as well (via the `onSelect` event)

https://stackblitz.com/edit/ui5wcr-vm-saveas-gewoqx?file=src%2FApp.tsx
